### PR TITLE
ARROW-2110: [Python] Only require pytest-runner on test commands

### PR DESCRIPTION
**Pull request by xhochy**

### CODE REVIEW

**Comment by xhochy** [comment]
cc @cpcloud

File: python/setup.py [hunk 1]
```diff
@@ -428,6 +428,13 @@ def parse_version(root):
     else:
         return version
 
+
+# Only include pytest-runner in setup_requires if we're invoking tests
+if {'pytest', 'test', 'ptr'}.intersection(sys.argv):
```
Comment (cpcloud) [inline python/setup.py]: What is `ptr`?

Comment (xhochy) [reply python/setup.py]: PyTestRunner

Comment (cpcloud) [reply python/setup.py]: pytest-runner i'm guessing

**Review by wesm — APPROVED** [review]
+1. Appveyor looking fine https://ci.appveyor.com/project/xhochy/arrow/build/1.0.561

**Pull request closed without merging**